app/main.py

```diff
@@ -1,5 +1,67 @@
+from __future__ import annotations
+
+
 class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self,
+                 name: str,
+                 weight: int | float,
+                 coords: list | float | None = None) -> None:
+
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords else [0, 0]
+
+    def go_forward(self, y_step: int | float = 1) -> None:
+        self.coords[1] += y_step
+
+    def go_back(self, y_step: int | float = 1) -> None:
+        self.coords[1] -= y_step
+
+    def go_right(self, x_step: int | float = 1) -> None:
+        self.coords[0] += x_step
+
+    def go_left(self, x_step: int | float = 1) -> None:
+        self.coords[0] -= x_step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self,
+                 name: str,
+                 weight: int | float,
+                 coords: list = None) -> None:
```

Comment:
Fix the type hint for `coords`
```suggestion
                 coords: list[int] | None = None) -> None:
```